samples/net/sockets/echo_client/README.rst

```diff
@@ -0,0 +1,63 @@
+.. _sockets-echo-client-sample:
+
+Echo Client
+###########
+
+Overview
+********
+
+The echo-client sample application for Zephyr implements a UDP/TCP client
+that will send IPv4 or IPv6 packets, wait for the data to be sent back,
+and then verify it matches the data that was sent.
+
+The source code for this sample application can be found at:
+:file:`samples/net/sockets/echo_client`.
+
+Requirements
+************
+
+- :ref:`networking_with_qemu`
+
+Building and Running
+********************
+
+There are multiple ways to use this application. One of the most common
+usage scenario is to run echo-client application inside QEMU. This is
+described in :ref:`networking_with_qemu`.
+
+Build echo-client sample application like this:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/net/sockets/echo_client
+   :board: <board_to_use>
+   :goals: build
+   :compact:
+
+``board_to_use`` defaults to ``qemu_x86``. In this case, you can run the
+application in QEMU using ``make run``. If you used another BOARD, you
+will need to consult its documentation for application deployment
+instructions. You can read about Zephyr support for specific boards in
+the documentation at :ref:`boards`.
+
+Running echo-server in Linux Host
+=================================
+
+There is one useful testing scenario that can be used with Linux host.
+Here echo-client is run in QEMU and echo-server is run in Linux host.
+
```

Comment:
As I mention in the server sample document, it would be worth cross-linking the server and client sample documents with something like this:

    See the :ref: `sockets-echo-server-sample` documentation for an alternate
    way of running, with the echo-client on the Linux host and the echo-server
    in QEMU.

Reply:
> As I mention in the server sample document

I don't see any comments there, do I miss something? Anyway I've applied simmilar cross-link there, I hope that was what you meant.

Reply:
Also, maybe it's worth to add link to native API variants of these sample (e.g. mention explicitly that socket samples are ports of those samples). (Yes, that's a lot of xrefs, but otherwise we risk people missing higher-level structure of samples, and the fact that we offer different implementations of the same functionalities, which may be useful info for people who want to compare and contrast APIs). Anyway, just a suggestion.

Reply:
Good point, I'll add the referance